utils/tests/test_appliance_cli.py

```diff
@@ -0,0 +1,71 @@
+from fixtures.pytest_store import store
+from utils import version
+from utils.appliance import provision_appliance
+from utils.conf import cfme_data, credentials
+import pytest
+
+
+def test_set_hostname(request):
+    store.current_appliance.ap_cli.set_hostname('test.example.com')
+    return_code, output = store.current_appliance.ssh_client.run_command(
+        "cat /etc/hosts | grep test.example.com")
+    assert return_code == 0
+
+
+@pytest.yield_fixture(scope="module")
+def provisioned_appliance():
+    ver_to_prov = str(version.current_version())
+    app = provision_appliance(version=ver_to_prov)
+
+    yield app
+
+    app.destroy()
+
+
+@pytest.fixture()
+def app_creds():
+    return {
+        'username': credentials['database']['username'],
+        'password': credentials['database']['password'],
+        'sshlogin': credentials['ssh']['username'],
+        'sshpass': credentials['ssh']['password']
+    }
+
+
+def test_configure_appliance_internal_fetch_key(request, provisioned_appliance, app_creds):
+        app = provisioned_appliance
+        fetch_key_ip = store.current_appliance.address
+        app.ap_cli.configure_appliance(0, 'localhost', app_creds['username'], app_creds['password'],
+            'vmdb_production', fetch_key_ip, app_creds['sshlogin'], app_creds['sshpass'])
+        assert app.is_evm_service_running()
+        app.ipapp.wait_for_web_ui()
+        assert app.is_web_ui_running()
+
+
+@pytest.fixture()
+def ipa_creds():
+    fqdn = cfme_data['ipa']['ipaserver']
+    fqdn.split('.', 1)
```

Comment:
oh yeh I missed that bit out, but your right the first line you mentioned is a lot nicer.